core/src/services/gcs/core.rs

```diff
@@ -193,6 +197,10 @@ impl GcsCore {
 
         let mut req = Request::get(&url);
 
+        if let Some(version) = args.version() {
+            req = req.header(constants::GENERATION, version);
```

Comment:
`generation` is a query parameter: https://cloud.google.com/storage/docs/json_api/v1/objects/get

Reply:
Yes, I misread that, it's been fixed.